src/lib.d

```diff
@@ -10,44 +10,47 @@ module ddmd.lib;
 
 import ddmd.globals;
 
-static if (TARGET_WINDOS)
+// Include all of these once these modules have been ported to all platforms.
+version (Windows)
 {
     import ddmd.libomf;
     import ddmd.libmscoff;
 }
-else static if (TARGET_LINUX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS)
+else version (OSX)
 {
-    import ddmd.libelf;
+    import ddmd.libmach;
 }
-else static if (TARGET_OSX)
+else version (Posix)
 {
-    import ddmd.libmach;
+    import ddmd.libelf;
 }
 else
 {
-    static assert(0, "unsupported system");
+    static assert(0, "OS not supported.");
 }
 
 extern (C++) class Library
 {
 public:
     static Library factory()
     {
-        static if (TARGET_WINDOS)
+        // This should be turned into runtime logic based on TARGET_* once
+        // these library generation modules are made cross-platform.
+        version (Windows)
         {
             return (global.params.mscoff || global.params.is64bit) ? LibMSCoff_factory() : LibOMF_factory();
         }
-        else static if (TARGET_LINUX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS)
```

Comment:
> It will enable cross-compilation someday, which is not possible with `static if`.

This change will never enable cross-compilation.  Other changes, that are not likely to be helped by this change, will do that.

> Nothing here is incorrect, as the exact same set of versions are used.

Not true.

> Except it's not cross-compiling yet, so the replacements in this file are just a hack, as I said. Would you like me to add a comment that this should be moved to `ddmd.target` someday?

No, I would like you to replace this change with something that adds value.